### ISSUE TITLE
why this xor pb in keras sometimes stop updating weight

### ISSUE THREAD

**Issue by khorsh-ed** [keras-team/keras#9742]
class` MyLayer(Layer):

    def __init__(self, output_dim, **kwargs):
     self.output_dim = output_dim
     super(MyLayer, self).__init__(**kwargs)

    def build(self, input_shape):
    # Create a trainable weight variable for this layer.
    # print("Weight : (", input_shape[1] ,",",self.output_dim,")")
     self.kernel = self.add_weight(name='kernel', 
                                  shape=(input_shape[1], self.output_dim),
                                  initializer='glorot_uniform',
                                  trainable=True)

    self.bias = self.add_weight(name='bias',
                            shape=(self.output_dim , 1 ),
                            initializer='zeros',
                            trainable=True)

    super(MyLayer, self).build(input_shape)  # Be sure to call this somewhere!

    def call(self, x):
    return K.sigmoid(K.dot(x, self.kernel) +self.bias )

    def compute_output_shape(self, input_shape):
    return (input_shape[0], self.output_dim)

   x_train = np.array([[0.0, 0.0], [0.0, 1.0], [1.0, 0.0], [1.0, 1.0]])
   y_train = np.array([[0], [1], [1], [0]])

    x = Input(shape=(2,))
    Hidden = Dense(3,activation='relu')(x)
    y =  MyLayer(output_dim=1)(Hidden)
    model = Model(inputs=x, outputs=y)
     model.compile(optimizer='adam',
          loss='binary_crossentropy',
          metrics=['accuracy'])
    model.fit(x_train, y_train, 
      batch_size=4, nb_epoch=10002, verbose=1)
